WPMedia/fusion-news-theme-blocks
# TMEDIA-585 - Add generator for content source block

## Description
Add a generator for content source blocks to boost themes developer productivity and provide a consistent way to create a starting point for a new content source block with all the boilerplate code and files

## Jira Ticket
- [TMEDIA-585](https://arcpublishing.atlassian.net/browse/TMEDIA-585)

## Test Steps

1. Checkout this branch `git checkout TMEDIA-584-content-source-generator`
2. Run `npm run generate:content-source`
3. Enter a name for your content source - `commerce`
4. Verify the correct files where created for the new content source block named "Commerce Content Source Block"
5. In feature pack repo add the following to the block array in the block.json file - `"@wpmedia/commerce-content-source-block@local",`
6. Start fusion `npx fusion start -f -l @wpmedia/commerce-content-source-block`
7. Verify the content source shows up in the debugger - http://localhost/pagebuilder/tools/debugger?_website=the-gazette&type=content-debugger

Note - Content source will not return any data in debugger as it's not hitting an Arc API endpoint

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
